### PR TITLE
Adding Schema property to manifest.

### LINKED ISSUE #54: Write JSON schema for Aspire manifest format.
We will be using the ```$schema``` attribute to specify the version of the manifest format we are using. That means we need a JSON schema which documents the format. We also need to think about where we host it.

### PR DESCRIPTION
**Pull request by mitchdenny**
Closes #54

Adding `$schema` property to generated manifests.

### CODE REVIEW

File: playground/AWS/AWS.AppHost/aspire-manifest.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+  "$schema": "https://json.schemastore.org/aspire-8.0.json",
```
Comment (danmoseley) [inline playground/AWS/AWS.AppHost/aspire-manifest.json:2]: noticed a typo on the schema itself: intepretted

File: playground/Stress/Stress.AppHost/aspire-manifest.json [hunk 2]
```diff
@@ -1,193 +1,195 @@
...
+        "OTEL_DOTNET_EXPERIMENTAL_OTLP_RETRY": "in_memory",
         "ASPNETCORE_FORWARDEDHEADERS_ENABLED": "true"
       },
       "bindings": {
         "http-5180": {
           "scheme": "http",
           "protocol": "tcp",
           "transport": "http",
-          "targetPort": 5180
+          "port": 5180
```
Comment (danmoseley) [inline playground/Stress/Stress.AppHost/aspire-manifest.json:18]: This doesn't represent a break that happened, it was just an error the schema caught?

File: playground/AspireEventHub/EventHubs.AppHost/aspire-manifest.json [hunk 3]
```diff
@@ -27,6 +28,7 @@
       "env": {
         "OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EXCEPTION_LOG_ATTRIBUTES": "true",
         "OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EVENT_LOG_ATTRIBUTES": "true",
+        "OTEL_DOTNET_EXPERIMENTAL_OTLP_RETRY": "in_memory",
```
Comment (danmoseley) [inline playground/AspireEventHub/EventHubs.AppHost/aspire-manifest.json:31]: Unrelated?

File: playground/PostgresEndToEnd/PostgresEndToEnd.AppHost/aspire-manifest.json [hunk 4]
```diff
@@ -1,9 +1,10 @@
 {
+  "$schema": "https://json.schemastore.org/aspire-8.0.json",
   "resources": {
     "pg1": {
       "type": "container.v0",
       "connectionString": "Host={pg1.bindings.tcp.host};Port={pg1.bindings.tcp.port};Username=postgres;Password={pg1-password.value}",
-      "image": "postgres:16.2",
+      "image": "docker.io/library/postgres:16.2",
```
Comment (danmoseley) [inline playground/PostgresEndToEnd/PostgresEndToEnd.AppHost/aspire-manifest.json:7]: Unrelated?

**Comment by danmoseley** [comment]
With respect to GA... how much do we expect users to edit the manifest manually?

**Comment by mitchdenny** [comment]
Well take this post GA.

File: playground/PostgresEndToEnd/PostgresEndToEnd.AppHost/aspire-manifest.json (hunk 4 above)
Comment (mitchdenny) [reply playground/PostgresEndToEnd/PostgresEndToEnd.AppHost/aspire-manifest.json:7]: Consequences of some other updates explicitly specifying registry use (to enable Podman). This manifest had not been regenerated since that change was made.

File: playground/AspireEventHub/EventHubs.AppHost/aspire-manifest.json (hunk 3 above)
Comment (mitchdenny) [reply playground/AspireEventHub/EventHubs.AppHost/aspire-manifest.json:31]: This manifest had not been regenerated since that change was made. I regenerated the manifests as part of this PR to keep them up to date. I uploaded all the playground manifests into the schemastore build system as postive test cases.

File: playground/Stress/Stress.AppHost/aspire-manifest.json (hunk 2 above)
Comment (mitchdenny) [reply playground/Stress/Stress.AppHost/aspire-manifest.json:18]: This was just a result of manifest regen.

**Comment by danmoseley** [comment]
Seems reasonable. Without this, schema validation (eg in VS Code) won't work, right? And assuming so, that can wait for 8.1?

**Comment by mitchdenny** [comment]
If the file is named `aspire-manifest.json` (due to VS and SchemaStore.org integration). But otherwise no, we won't get intelligence in VS.

But I think that is OK for 8.0. This PR is to get it into 8.1 anyway.

File: src/Shared/SchemaUtils.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.Utils;
+
+internal static class SchemaUtils
+{
+    public const string SchemaVersion = "https://json.schemastore.org/aspire-8.0.json";
```
Comment (danmoseley) [inline src/Shared/SchemaUtils.cs:8]: SchemaUrl ?

**Review by danmoseley — APPROVED** [review]
